File: VB/BootstrapThemeSwitcher.ascx.vb
Option Infer On

Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls

Partial Public Class BootstrapThemeSwitcher
    Inherits UserControl

    Public Const DefaultTheme As String = "default"
    Public Const CurrentThemeCookieKey As String = "DXBSCurrentTheme"


    Private themeLinkID_Renamed As String

    Public Property ThemeLinkID() As String
        Get
            Return HiddenLink("themeLinkID").ToString()
        End Get
        Set(ByVal value As String)
            HiddenLink("themeLinkID") = value
        End Set
    End Property


    Public ReadOnly Property CurrentTheme() As String
        Get
            If HttpContext.Current.Request.Cookies(CurrentThemeCookieKey) IsNot Nothing Then
                Return HttpContext.Current.Request.Cookies(CurrentThemeCookieKey).Value
            End If
            Return DefaultTheme
        End Get
    End Property

    Protected Sub Page_Init(ByVal sender As Object, ByVal e As EventArgs)
        FillThemesMenu()

    End Sub

    Protected Sub FillThemesMenu()

        Dim currentTheme_Renamed = CurrentTheme

        For Each theme_Renamed In BootstrapThemesModel.Current.Themes
            Dim item = themesMenu.Items.Add(theme_Renamed.Title, theme_Renamed.Name)
            item.IconCssClass = theme_Renamed.Name
            If theme_Renamed.Name = currentTheme_Renamed Then
                item.Selected = True
            End If
        Next theme_Renamed
    End Sub
End Class

File: CS/BootstrapThemeSwitcher.ascx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;

public partial class BootstrapThemeSwitcher : UserControl {

    public const string DefaultTheme = "default";
    public const string CurrentThemeCookieKey = "DXBSCurrentTheme";

    private string themeLinkID;

    public string ThemeLinkID {
        get { return HiddenLink["themeLinkID"].ToString(); }
        set {
            HiddenLink["themeLinkID"] = value;
        }
    }


    public string CurrentTheme {
        get {
            if(HttpContext.Current.Request.Cookies[CurrentThemeCookieKey] != null)
                return HttpContext.Current.Request.Cookies[CurrentThemeCookieKey].Value;
            return DefaultTheme;
        }
    }

    protected void Page_Init(object sender, EventArgs e) {
        FillThemesMenu();
        
    }

    protected void FillThemesMenu() {
        var currentTheme = CurrentTheme;
        foreach(var theme in BootstrapThemesModel.Current.Themes) {
            var item = themesMenu.Items.Add(theme.Title, theme.Name);
            item.IconCssClass = theme.Name;
            if(theme.Name == currentTheme)
                item.Selected = true;
        }
    }
}